### PR TITLE
Fix `TestAccuracy`: Randomly reduce testing parameters

### LINKED ISSUE #7819: Flaky test: `chainer_tests/functions_tests/evaluation_tests/test_accuracy.py`
https://ci.preferred.jp/chainer.py27and35.gpu/20768/

Timeout due to the increased number of test variations. Should consider reducing number of tests.

### PR DESCRIPTION
**Pull request by toslunar**
Close #7819.

It's unlikely for `TestAccuracy` to fail with only a few testing parameters, because the implementation of `F.accuracy` is not complicated.

### CODE REVIEW

**Comment by niboshi** [comment]
I think random selection is too crude. Important combination could possibly be dropped.
We can instead manually reduce the combination.

**Comment by toslunar** [comment]
Now it's guaranteed that all pairs are tested.  While the algorithm is naive but it only generates 40 cases out of 6×3×3×3=162.

File: tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py [hunk 1]
```diff
@@ -12,6 +13,34 @@
 from chainer.utils import type_check
 
 
+def _testing_pairwise(*parameters):
```
Comment (niboshi) [inline tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py]: I found this function difficult to understand.
Especially it's difficult to track the data structures of each variables (e.g. `parameters`, `product_dicts`, etc.).
Could you annotate the code with approppriate comments/assertions?

File: tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py [hunk 2]
```diff
@@ -12,6 +13,34 @@
...
+    parameters = [list(enumerate(dicts)) for dicts in parameters]
+
+    uncovered = {}
+    for (i, dicts_i), (j, dicts_j) in itertools.combinations(
+            enumerate(parameters), 2):
+        uncovered[(i, j)] = set(itertools.product(
+            range(len(dicts_i)), range(len(dicts_j))))
+
+    product_dicts = list(itertools.product(*parameters))
+    rs.shuffle(product_dicts)
```
Comment (niboshi) [inline tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py]: Why is this shuffle required?

File: tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py (hunk 1 above)
Comment (toslunar) [reply tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py]: Sure.  I'll add comments or assertions.

File: tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py (hunk 2 above)
Comment (toslunar) [reply tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py]: The shuffle reduces the number of tests.  The following improvements may address your concern.
- sort selected tests, and/or
- output indices as they were generated by `testing.parameterize`

**Comment by toslunar** [comment]
The priority is lowered because #7819 is closed.

**Pull request merged**